### PR TITLE
(GH-132) List command

### LINKED ISSUE #132: [API] Expose Package Results through the API

### LINKED ISSUE #223: choco list --localonly doesn't work from API
The configuration.LocalOnly is being handled only during commandline parsing, not during execution.

### PR DESCRIPTION
**Pull request by Jaykul**
This set of changes fixes the various incarnations of the LIST command so that they return ... lists :-)
It depends on #224 

Closes #132
Closes #223

### CODE REVIEW

File: src/chocolatey/infrastructure.app/services/NugetService.cs [hunk 1]
```diff
@@ -895,14 +889,13 @@ private void set_package_names_if_all_is_specified(ChocolateyConfiguration confi
                 config.PackageNames = string.Empty;
                 var input = config.Input;
                 config.Input = string.Empty;
-
-                var localPackages = list_run(config, logResults: false);
-
+	            config.Quiet = true;
+				
+                config.PackageNames = list_run(config).Select(p => p.Name).@join(ApplicationParameters.PackageNamesSeparator);
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/services/NugetService.cs]: Don't leave my config set to quiet.

Comment (Jaykul) [inline src/chocolatey/infrastructure.app/services/NugetService.cs]: Good point

**Comment by RichiCoder1** [comment]
@Jaykul @ferventcoder Is there anything that needs to happen for these to be unblocked? Really would like to incorporate these for ChocoGUI

**Comment by ferventcoder** [comment]
I think I just need to take a look here and pull this in.

File: src/chocolatey/infrastructure.app/commands/ChocolateyListCommand.cs [hunk 2]
```diff
@@ -16,17 +16,19 @@
...
     using domain;
     using infrastructure.commands;
     using logging;
+    using results;
     using services;
 
     [CommandFor(CommandNameType.list)]
     [CommandFor(CommandNameType.search)]
-    public sealed class ChocolateyListCommand : ICommand
+    public sealed class ChocolateyListCommand : IListCommand<PackageResult>
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/commands/ChocolateyListCommand.cs]: @Jaykul I don't see `IListCommand<T>` at all.

Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/commands/ChocolateyListCommand.cs]: I ended up reimplementing it, but didn't constrain `T` as I am unsure how you would have constrained it.

Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/commands/ChocolateyListCommand.cs]: Found that it was already in master only.

**Comment by ferventcoder** [comment]
Merged into stable at https://github.com/chocolatey/choco/commit/3c12763a438cc19dd758bd47f07383b7bc13de86 and master at https://github.com/chocolatey/choco/commit/909ab1830f9e4abfcae80959596ce2b60b76ed77

**Pull request closed without merging**